### PR TITLE
[#3761 Issue] Add sorting feature to MetricRegistryMetricReader

### PR DESCRIPTION
**Pull request by alexVengrovsk**
The original issue was folowing:
```MetricRegistryMetricReader``` uses simple ```HashSet``` to collect metrics from registry. As result they appear it unpredictable order when exported to ```web/jmx``` view. It would be expected to see metrics ordered by name instead of
```
dw.a.run.snapshot.95thPercentile: 10,
dw.a.run.expexp.snapshot.99thPercentile: 82,
dw.a.run.dump.snapshot.max: 1,
dw.a.run.expexp.snapshot.95thPercentile: 9,
dw.a.run.dump.snapshot.999thPercentile: 1,
dw.a.run.count: 2,
```
Improvements in this PR add sorting fetures to the ```findAll()``` method of ```MetricRegistryMetricReader``` class by using ```TreeSet(Comparator<? super E> comparator)``` insted of  ```HashSet()```.

### CODE REVIEW

File: spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/reader/MetricRegistryMetricReader.java [hunk 1]
```diff
@@ -110,7 +112,12 @@ public MetricRegistryMetricReader(MetricRegistry registry) {
 		return new Iterable<Metric<?>>() {
 			@Override
 			public Iterator<Metric<?>> iterator() {
-				Set<Metric<?>> metrics = new HashSet<Metric<?>>();
+				Set<Metric<?>> metrics = new TreeSet<Metric<?>>(new Comparator<Metric<?>>() {
+					@Override
+					public int compare(Metric<?> metric1, Metric<?> metric2) {
+						return metric2.getName().compareToIgnoreCase(metric2.getName());
```
Comment (spencergibb) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/reader/MetricRegistryMetricReader.java]: `metric1` first? you have `metric2` twice.

Comment (alexVengrovsk) [inline spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/reader/MetricRegistryMetricReader.java]: Thank you for help

**Comment by wilkinsona** [comment]
Thanks for the PR.

I'm not sure that `MetricRegistryMetricReader` is the right place to apply the ordering. It would mean that everyone pays the cost of ordering, even if they don't care about the order. The original issue talked about web and JMX requests; a more appropriate place to apply ordering might be in `MetricsEndpoint` and, even then, I'm not sure that we'd want to order the result by default.

**Comment by alexVengrovsk** [comment]
Yes. You are right. The cost will be high, and if it is not appropriate change, the PR should be closed. I will do it.

**Pull request closed without merging**